[Uploadable] Fixes issue with metadata when uploading entities of different types #2071

JoolsMcFly opened the pull request:
[Uploadable]
If one uploads two entities of different classes then only meta for the first class is loaded which causes `getFilePathFieldValue` to complain because the entity does not match the metadata.

This PR ensures we get the metadata corresponding to the entity being uploaded.

JoolsMcFly commented:
@phansys I see you've added `Needs Changelog note`. Do you want me to take care of this or is it something maintainers do?

I can change
```
## [Unreleased]

### Added
- Tree: Add `Nested::ALLOWED_NODE_POSITIONS` constant in order to expose the available node positions

### Fixed
- Sortable: Fix return value check of Comparable interface (#2541)
```

to 
```
## [Unreleased]

### Added
- Tree: Add `Nested::ALLOWED_NODE_POSITIONS` constant in order to expose the available node positions

### Fixed
- Sortable: Fix return value check of Comparable interface (#2541)
- Uploadable: Retrieve correct metadata when uploading entities of different classes (#2071)
```

Would that work?

phansys commented:
Yes @JoolsMcFly, that would be great!
If you have any doubts, please check [`CONTRIBUTING.md`](https://github.com/doctrine-extensions/DoctrineExtensions/blob/c9f9b1fa1e7bde4016c7e5a0e8741eee4a3f5107/CONTRIBUTING.md#changelog).

JoolsMcFly commented:
Yeah, sorry, I skimmed through CONTRIBUTING.md without paying attention to changelog! D'oh!

Just committed changelog updates.

JoolsMcFly commented:
Hi. I see `Needs test` label was added. Do you want me to write tests or is the label aimed at maintainers?
Thanks!

phansys commented:
Any feature or fix SHOULD include tests in order to ensure the behavior and avoid regressions. If you can, it would be great if you would add some use cases covering this change.

JoolsMcFly commented:
@phansys I've added a test. Prior to my fix the following error was found upon second `flush`:
>Error: Cannot access private property Gedmo\Tests\Uploadable\Fixture\Entity\Image::$filePath

With my fix test runs fine. I added simple assertions since we're more interested in not getting an error than checking filepaths (which is already covered in other tests).

WDYT?

phansys reviewed (approved):
LGTM.

franmomu commented:
thanks @JoolsMcFly!

JoolsMcFly commented:
You're welcome!
